doc/sjs-for-js/es6-to-scala-part3.md

```diff
@@ -260,6 +260,234 @@ Here we use triple-quoted strings that allow us to write regex without escaping
 converted into a {% scaladoc util.matching.Regex %} object with the `.r` method. Because regexes extract strings, we need
 to convert matched groups to integers ourselves.
 
+## Functions revisited
+
+We covered the basic use functions in [Part 1](es6-to-scala-part1.html), but Scala, being a functional programming
+language, provides much more when it comes to functions. Let's explore some of the more advanced features and how they
+compare to JavaScript.
+
+#### Higher-order functions
+
+Scala allows the definition of higher-order functions. These are functions that take other functions as parameters, or
+whose result is a function. Higher-order functions should be familiar to JavaScript developers, because they often
+appear in form of functions that take callbacks as parameters.
+
+Typically higher-order functions are used to pass specific functionality to a general function, like in the case of
+`Array.prototype.filter` in ES6 or `Seq.filter` in Scala. We can use this to build a function to calculate an average
+from a sequence of values, using a function to extract the target value.
+
+{% columns %}
+{% column 6 ES6 %}
+{% highlight javascript %}
+function avgBy(arr, f) {
+  const sum = arr.reduce((sum, e) => sum + f(e), 0);
+  return sum/arr.length;
+}
+
+const avgAge = avgBy(persons, e => e.age);
+{% endhighlight %}
+{% endcolumn %}
+
+{% column 6 Scala %}
+{% highlight scala %}
+def avgBy[T](seq: Seq[T], f: T => Int): Double = {
+  val sum = seq.foldLeft(0)((sum, e) => sum + f(e))
+  sum.toDouble/seq.length
+}
+
+val avgAge = avgBy[Person](persons, _.age)
+{% endhighlight %}
+{% endcolumn %}
+{% endcolumns %}
+
+#### Call-by-Name
```

Comment:
I think this is too much. I love call-by-name parameters, but I feel like they live in an even more advanced space than the other features discussed on this page.